cmd/csi-driver/csi-driver.go

```diff
@@ -51,6 +52,18 @@ var (
 			} else {
 				log.InitLogging(csiControllerLogFile, nil, true)
 			}
+			// Scanning for stale entries on each node where CSI resides, 
+			// this check makes sure that it runs only on hpe-csi-nodes pods 
+			podName := os.Getenv("POD_NAME")
+			if strings.Contains(podName, "hpe-csi-node-") { //&& !strings.Contains(podName, "hpe-csi-controller-") {
+        	                log.Info("********Cleaning the stale entries before starting the HPE CSI driver********")
+				log.Info("Running rescan-scsi-bus.sh script to flush the stale multipath devices...")
+				output, _, err := util.ExecCommandOutputWithTimeout("rescan-scsi-bus.sh", []string{"-f", "-m", "-r"}, 120)
+				if err != nil {
+					log.Error("unable to flush the stale multipath devices using rescan-scsi-bus.sh script")
+				}
```

Comment:
If this fails, we need to understand why and determine if it's fatal. And if it succeeds, we need to ensure the node is still not unhealthy. If any of those two conditions are met, reboot.